checks.d/mysql.py

```diff
@@ -535,17 +535,17 @@ def _collect_metrics(self, host, db, tags, options, queries):
             # slaves will only be collected iff user has PROCESS privileges.
             slaves = self._collect_scalar('Slaves_connected', results)
 
-            if slave_running is not None:
+            if slave_running and not slaves:  # slave
                 if slave_running.lower().strip() == 'on':
                     slave_running_status = AgentCheck.OK
                 else:
                     slave_running_status = AgentCheck.CRITICAL
-            elif slaves or binlog_running:
-                if slaves and slaves > 0 and binlog_running:
+            elif slaves or binlog_running:  # master
```

Comment:
I think it's fine if we do the comparison below. Also, it looks like float(0) resolves to False on `2.7.11` anyways.